src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1820,15 +1880,145 @@ class NodeUtils(object):
+            if 'gis' not in gpus[gpuid]:
+                gpus[gpuid]['gis'] = {}
+            gpus[gpuid]['gis'][giid] = gi
+
+        # now find all CIs
+        cmd = [self.cfg['nvidia-smi'], 'mig', '-lci']
+        pbs.logmsg(pbs.EVENT_DEBUG4, 'NVIDIA SMI command: %s' % cmd)
+        time_start = time.time()
+        out = []
+        try:
+            # Try running the nvidia-smi command
+            process = subprocess.Popen(cmd, shell=False,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE,
+                                       universal_newlines=True)
+            out = process.communicate()[0].split('\n')
+        except Exception:
+            pbs.logmsg(pbs.EVENT_DEBUG4, 'Failed to execute: %s' %
+                       " ".join(cmd))
+            pbs.logmsg(pbs.EVENT_DEBUG3, '%s: No MIGs found' % caller_name())
+            return
+        elapsed_time = time.time() - time_start
+        if elapsed_time > 2.0:
+            pbs.logmsg(pbs.EVENT_DEBUG,
+                       '%s: nvidia-smi call took %f seconds' %
+                       (caller_name(), elapsed_time))
+
+        # format is
+        # | Compute instances:                                    |
+        # | GPU     GPU       Name             Profile   Instance |
+        # |       Instance                       ID        ID     |
+        # |         ID                                            |
+        # |   0      7       MIG 1g.5gb           0         0     |
+        r = re.compile(
+            r'^\|\s+(\d+)\s+(\d+)\s+MIG\s+\S+\s+\d+\s+(\d+)\s+(\S+\s+)?\|$')
+        for l in out:
+            match = r.match(l)
+            if not match:
+                continue
+            gpu_num = int(match.group(1))
+            gpuid = 'nvidia%d' % gpu_num
+            giid = int(match.group(2))
+            ciid = int(match.group(3))
+            minor = self._discover_mig_minor(gpu_num, giid, ciid)
+            if minor == -1:
+                continue
+            uuid = 'MIG-%s/%s/%s' % (gpus[gpuid]['uuid'], giid, ciid)
+            ci = {'minor': minor, 'gi': giid, 'ci': ciid, 'uuid': uuid}
+            pbs.logmsg(pbs.EVENT_DEBUG4, 'CI found %s' % str(ci))
+            if 'cis' not in gpus[gpuid]['gis'][giid]:
+                gpus[gpuid]['gis'][giid]['cis'] = {}
+            gpus[gpuid]['gis'][giid]['cis'][ciid] = ci
+        return
+
+    def _discover_mig_minor(self, gpu, gi, ci=None):
+        pbs.logmsg(pbs.EVENT_DEBUG4, '%s: Method called' % caller_name())
+        path = os.path.join(os.sep, 'proc', 'driver',
+                            'nvidia-caps', 'mig-minors')
+        if ci is None:
+            capname = 'gpu%d/gi%d/access' % (gpu, gi)
+        else:
+            capname = 'gpu%d/gi%d/ci%d/access' % (gpu, gi, ci)
+        with open(path, 'r') as minors:
+            for line in minors:
+                cap, minor = line.split()
```

Comment:
If the format of the proc file changes and you get back more than 2 columns, this will barf.  How about line.split(' ', 1)?

Reply:
This is a documented, supported interface by nvidia, it shouldn't change without notice. Because of this, I don't think we need to change it.

Reply:
Well just incase it does change, I'd rather us do split(' ', 1) because then we'll fail to query the MIG and not raise an exception.  We'll have to change the hook in either case, but at least this way it won't stop working.